src/rpc/util.h

```diff
@@ -169,30 +175,25 @@ struct RPCArg {
     using DefaultHint = std::string;
     using Default = UniValue;
     using Fallback = std::variant<Optional, /* hint for default value */ DefaultHint, /* default constant value */ Default>;
+
     const std::string m_names; //!< The name of the arg (can be empty for inner args, can contain multiple aliases separated by | for named request arguments)
     const Type m_type;
-    const bool m_hidden;
     const std::vector<RPCArg> m_inner; //!< Only used for arrays or dicts
     const Fallback m_fallback;
     const std::string m_description;
-    const std::string m_oneline_description; //!< Should be empty unless it is supposed to override the auto-generated summary line
-    const std::vector<std::string> m_type_str; //!< Should be empty unless it is supposed to override the auto-generated type strings. Vector length is either 0 or 2, m_type_str.at(0) will override the type of the value in a key-value pair, m_type_str.at(1) will override the type in the argument description.
+    const RPCArgOptions m_opts;
 
     RPCArg(
         const std::string name,
         const Type type,
         const Fallback fallback,
         const std::string description,
-        const std::string oneline_description = "",
-        const std::vector<std::string> type_str = {},
-        const bool hidden = false)
+        RPCArgOptions opts = {})
```

Comment:
Since these are always (I think?) rvalue refs, could avoid copying? + for the other ctor
```suggestion
        RPCArgOptions&& opts = {})
```

Reply:
I don't think adding `&&` here will avoid a copy. `&&` is just a reference type that forces the caller to pass in an rvalue(ref). This would only avoid a copy if it was an lvalue, cast to an rvalue. Though, in that case we might want to re-use the lvalue and not force the caller to move it (or explicitly copy it), so I think `RPCArgOptions` is fine here.

While both are fine here, a pure r-value and an r-value ref, I think it is fine to only use `&&` in cases where a copy can be avoided and the copy would be costly (e.g. a8f69541ad53a76d5f69044ba829069d225a4af1)

Reply:
I think you're right that there is no need for `&&` here, but I think the reason is that since c++17 [mandatory copy elision](https://en.cppreference.com/w/cpp/language/copy_elision) avoids copying the temporary in the first place? 

Without copy elision, I think passing by value instead of rvalue ref would have required a constructing and then copy-constructing operation, as pass by value always means copying (except with elision)?

Drifting off-topic, so you can consider this resolved. Don't want to waste anyone's time.

Reply:
I guess you are right that for arguments copy elision is not mandatory, but I'd be surprised to see a compiler that wouldn't do it voluntarily.

Though, happy to change to `&&`. If other reviewers see this they can use :-1: and :+1: